Mathlib/Analysis/Normed/Group/SeparationQuotient.lean

```diff
@@ -0,0 +1,161 @@
+/-
+Copyright (c) 2024 Yoh Tanimoto. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yoh Tanimoto
+-/
+import Mathlib.Analysis.Normed.Group.Hom
+
+/-!
+# Lifts of maps to separation quotients of seminormed groups
+
+For any `SeminormedAddCommGroup M`, a `NormedAddCommGroup` instance has been defined in
+`Mathlib.Analysis.Normed.Group.Uniform`.
+
+## Main definitions
+
+We use `M` and `N` to denote seminormed groups.
+All the following definitions are in the `SeparationQuotient` namespace. Hence we can access
+`SeparationQuotient.normedMk` as `normedMk`.
+
+* `normedMk` : the normed group hom from `M` to `SeparationQuotient M`.
+
+* `liftNormedAddGroupHom` : implements the universal property of `SeparationQuotient M`. Here
+    `(f : NormedAddGroupHom M N)`, `(hf : ∀ x : M, ‖x‖ = 0 → f x = 0)` and
+    `liftNormedAddGroupHom f hf : NormedAddGroupHom (SeparationQuotient M) N`.
+
+## Main results
+
+* `norm_normedMk` : the operator norm of the projection is `1` if the subspace is not dense.
+
+* `norm_liftNormedAddGroupHom_le` : `‖liftNormedAddGroupHom f hf‖ ≤ ‖f‖`.
+
+## Implementation details
+
+For any `SeminormedAddCommGroup M`, we define a norm on `SeparationQuotient M` by
+`‖x‖ = ‖mk x‖` using the lift.
+
+-/
+
+
+noncomputable section
+
+open SeparationQuotient NNReal
+
+variable {M N : Type*} [SeminormedAddCommGroup M] [SeminormedAddCommGroup N]
+
+namespace SeparationQuotient
+
+/-- For `(m : M)`, `mk m = 0` if and only if `‖m‖ = 0`. -/
+theorem mk_eq_zero_iff (m : M) : mk m = 0 ↔ ‖m‖ = 0 := by
```

Comment:
When the docstring is identical to the theorem statement, we should just not have a docstring:
```suggestion
theorem mk_eq_zero_iff (m : M) : mk m = 0 ↔ ‖m‖ = 0 := by
```